### PR TITLE
Add `onDidAnyChange` method

### LINKED ISSUE #45: On change for the whole config
Hello! Would it be possible to have a `.onChange` handler for the whole config object?

### PR DESCRIPTION
**Pull request by dalgleish**
Fix #45: Hi this is my first contribution so I just attempted to follow the existing convention.
Please let me know if there are any corner cases I didn't consider. I ran tests with node `v6.16.0`, `v8.14.0`, `v10.15.0`.

I moved `isEqual` to a static method and reused it in both `onDidChange` and `onDidAnyChange`.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -95,6 +95,39 @@ class Conf {
...
+		const onChange = () => {
+			const oldValue = currentValue;
+			const newValue = this.store;
+
+			const didChange = Conf.isEqual(oldValue, newValue) === false;
+			if (didChange) {
+				currentValue = newValue;
+				callback.call(this, newValue, oldValue);
+			}
+		};
```
Comment (dalgleish) [inline index.js]: `onChange` repeats some logic as in `onDidChange`, maybe it could be generalized.

File: readme.md [hunk 2]
```diff
@@ -145,6 +145,12 @@ Delete all items.
 
 Watches the given `key`, calling `callback` on any changes. When a key is first set `oldValue` will be `undefined`, and when a key is deleted `newValue` will be `undefined`.
 
+#### .onDidAnyChange(callback)
+
+`callback`: `(newValue, oldValue) => {}`
+
+Calls `callback` on changes anywhere in the config file. `oldValue` and `newValue` will be the full config object before after the change, respectively. You must compare `oldValue` to `newValue` to find out what changed.
```
Comment (brandon93s) [inline readme.md]: before *and* after

File: index.js [hunk 3]
```diff
@@ -95,6 +95,39 @@ class Conf {
 		this.store = plainObject();
 	}
 
+	static isEqual(newValue, oldValue) {
```
Comment (brandon93s) [inline index.js]: This makes it unnecessarily available on the public interface to consumers. This could just be a private helper method in the file, similar to `plainObject` to prevent unintentional usage.

File: index.js (hunk 1 above)
Comment (brandon93s) [reply index.js]: [dot-prop](https://github.com/sindresorhus/dot-prop) supports [returning the entire object](https://github.com/sindresorhus/dot-prop/blob/42cc3bf8ec8a404bca121a70ea831a9d687582a2/index.js#L24-L26) when a path is excluded:

```js
	get(obj, path, value) {
		if (!isObj(obj) || typeof path !== 'string') {
			return value === undefined ? obj : value;
		}
```

This should pave the path to greatly simplify the two implementations since `onDidAnyChange` will just be `onDidChange` with an empty `key`. To support separate validation, the internals could be extracted into a shared method.

File: index.js (hunk 3 above)
Comment (brandon93s) [reply index.js]: If refactored into a shared method, extraction of the equal logic may be unnecessary.

**Comment by sindresorhus** [comment]
@dalgleish Bump :)

**Comment by sindresorhus** [comment]
@dalgleish Bump2
